### DISCUSSION TITLE
[problem] Install the Testsigma by docker-compose, happened problem stay at mysql container staring long time

### DISCUSSION THREAD

**Discussion by tiandong19860806** [testsigmahq/testsigma#79] [General]
Hi Testsigma teams,

I have a problem that I installed the Testsigma by docker-compose, happened problem stay at mysql container staring long time.

detail as below:

And I expect result : it is start testsigma two container normal and not stay starting long time.

Thanks.

(1) OS version
[root@CNT7XATST01 ~]# uname -a
Linux CNT7XATST01 3.10.0-1160.el7.x86_64 #1 SMP Mon Oct 19 16:18:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

(2) docker version
[root@CNT7XATST01 ~]# docker --version
Docker version 20.10.17, build 100c701
[root@CNT7XATST01 ~]# 
[root@CNT7XATST01 ~]# docker-compose --version
Docker Compose version v2.9.0

(3) Testsigma install by docker as below
cat > /home/app/testsigma-server/docker-compose-yml/docker-compose.yml << EOF
version: "3.9"
services:
  mysql:
    image: mysql:5.7
    platform: linux/x86_64
    container_name: testsigma_mysql
    restart: always
    volumes:
      - /home/data/testsigma-server/db_data:/var/lib/mysql
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: db_testsigma_opensource
      LANG: C.UTF-8
      LC_ALL: C.UTF-8
    ports:
      - "3306:3306"
    command: --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci
    healthcheck:
      test: out=$$(mysqladmin ping -h localhost -P 3306 -u root --password=root 2>&1); echo $$out | grep 'mysqld is alive' || { echo $$out; exit 1; }
      interval: 300s
      retries: 10
  testsigma_server:
    image: testsigmahq/server:v1.1.0
    container_name: testsigma_server
    ports:
      - "9090:9090"
      - "443:443"
    volumes:
      - /home/data/testsigma-server/ts_data:/opt/app/ts_data
    depends_on:
      mysql:
        condition: service_healthy
EOF
-- and detail steps , look attachment

(4) problem as below:
**pic 1**: start up docker-compose, **the mysql container is long time stay  starting**.
![图片](https://user-images.githubusercontent.com/12596666/182788845-f16f17bf-6a16-494d-97a1-30a08010d52a.png)

but the docker images , had compeleted download ago.
![图片](https://user-images.githubusercontent.com/12596666/182789052-8722e559-391c-4899-8b88-a5efcd8364de.png)

pic 2: but actual ,the container [mysql] had completed completed, I can connect it by two case.
(1) connect mysql container by outside it
![图片](https://user-images.githubusercontent.com/12596666/182789845-5ea01f88-accc-448b-ae4c-8f0aab3c5d86.png)

(2) connect mysql container by inside it
![图片](https://user-images.githubusercontent.com/12596666/182790565-498759ae-1fa9-42b7-818e-132f048c6512.png)

**Comment by tiandong19860806** [comment]
-- step 1
-- 创建软件安装目录
mkdir -p /home/app/testsigma-server/ && \
mkdir -p /home/log/testsigma-server/ && \
mkdir -p /home/data/testsigma-server/
-- mkdir -p /home/data/testsigma-server/db_data && \
-- mkdir -p /home/data/testsigma-server/ts_data

chmod -R 777 /home/app/testsigma-server/ && \
chmod -R 777 /home/log/testsigma-server/ && \
chmod -R 777 /home/data/testsigma-server/ && \
chown -R docker:docker /home/app/testsigma-server/ && \
chown -R docker:docker /home/log/testsigma-server/ && \
chown -R docker:docker /home/data/testsigma-server/

-- ============================================================================
-- ============================================================================

-- step 2
-- 创建配置文件
-- 参考网址：https://github.com/testsigmahq/testsigma/blob/dev/deploy/docker/docker-compose.yml

mkdir -p /home/app/testsigma-server/docker-compose-yml/
  
cat > /home/app/testsigma-server/docker-compose-yml/docker-compose.yml << EOF
version: "3.9"
services:
  mysql:
    image: mysql:5.7
    platform: linux/x86_64
    container_name: testsigma_mysql
    restart: always
    volumes:
      - /home/data/testsigma-server/db_data:/var/lib/mysql
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: db_testsigma_opensource
      LANG: C.UTF-8
      LC_ALL: C.UTF-8
    ports:
      - "3306:3306"
    command: --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci
    healthcheck:
      test: out=$$(mysqladmin ping -h localhost -P 3306 -u root --password=root 2>&1); echo $$out | grep 'mysqld is alive' || { echo $$out; exit 1; }
      interval: 300s
      retries: 10
  testsigma_server:
    image: testsigmahq/server:v1.1.0
    container_name: testsigma_server
    ports:
      - "9090:9090"
      - "443:443"
    volumes:
      - /home/data/testsigma-server/ts_data:/opt/app/ts_data
    depends_on:
      mysql:
        condition: service_healthy
EOF

-- ============================================================================
-- ============================================================================

-- step 3
-- 启动软件 testsigma-server 服务端应用

-- 参考资料：
-- docker-compose up -d 和 restart的区别
-- https://blog.csdn.net/u010953880/article/details/119956755
-- https://www.jianshu.com/p/c4285ad1572d 


-- ============================================================================

-- 切换账号
su - docker 

-- ============================================================================

cd /home/app/testsigma-server/docker-compose-yml/
 
-- 安装和启动软件
-- -d 终端显示屏不打印日志
docker-compose up -d
-- 附录：停止软件：docker-compose stop


-- 执行结果，如下：
#-- [root@CNT7XATST01 docker-compose]-- docker-compose up -d
#-- [+] Running 33/33
#--  ⠿ mysql Pulled                                                                                                                                                                                         203.7s
#--    ⠿ 72a69066d2fe Pull complete                                                                                                                                                                          47.3s
#--    ⠿ 93619dbc5b36 Pull complete                                                                                                                                                                          47.5s
#--    ⠿ 99da31dd6142 Pull complete                                                                                                                                                                          48.4s
#--    ⠿ 626033c43d70 Pull complete                                                                                                                                                                          48.9s
#--    ⠿ 37d5d7efb64e Pull complete                                                                                                                                                                          49.1s
#--    ⠿ ac563158d721 Pull complete                                                                                                                                                                          52.9s
#--    ⠿ d2ba16033dad Pull complete                                                                                                                                                                          53.2s
#--    ⠿ 0ceb82207cd7 Pull complete                                                                                                                                                                          53.3s
#--    ⠿ 37f2405cae96 Pull complete                                                                                                                                                                         184.6s
#--    ⠿ e2482e017e53 Pull complete                                                                                                                                                                         185.0s
#--    ⠿ 70deed891d42 Pull complete                                                                                                                                                                         185.1s
#--  ⠿ testsigma_server Pulled                                                                                                                                                                              316.2s
#--    ⠿ 2d473b07cdd5 Pull complete                                                                                                                                                                         193.1s
#--    ⠿ 033ec090831e Pull complete                                                                                                                                                                         193.3s
#--    ⠿ 568c453653bc Pull complete                                                                                                                                                                         193.5s
#--    ⠿ 0ed6a10c6abf Pull complete                                                                                                                                                                         205.4s
#--    ⠿ a289d18e7958 Pull complete                                                                                                                                                                         207.2s
#--    ⠿ 1ab51add53b5 Pull complete                                                                                                                                                                         209.6s
#--    ⠿ 424cd4bb4a59 Pull complete                                                                                                                                                                         234.4s
#--    ⠿ 01f1b3c270c6 Pull complete                                                                                                                                                                         234.6s
#--    ⠿ def109320a2a Pull complete                                                                                                                                                                         234.8s
#--    ⠿ 09d29ce3d2ff Pull complete                                                                                                                                                                         235.1s
#--    ⠿ e0a925c7d85f Pull complete                                                                                                                                                                         235.3s
#--    ⠿ b2cc15ed027c Pull complete                                                                                                                                                                         235.4s
#--    ⠿ d0db3246abb0 Pull complete                                                                                                                                                                         235.6s
#--    ⠿ d47289fbbc58 Pull complete                                                                                                                                                                         239.7s
#--    ⠿ d7405b9e9196 Pull complete                                                                                                                                                                         240.2s
#--    ⠿ 513b931ff86b Pull complete                                                                                                                                                                         246.5s
#--    ⠿ ca7fd65e5cfe Pull complete                                                                                                                                                                         246.7s
#--    ⠿ 2026f94b951d Pull complete                                                                                                                                                                         247.0s
#--    ⠿ a206e6c6a980 Pull complete                                                                                                                                                                         247.2s
#--    ⠿ 6ee28eab9048 Pull complete                                                                                                                                                                         247.3s
#-- [+] Running 2/3
#--  ⠿ Network docker-compose_default  Created                                                                                                                                                                0.2s
#--  ⠿ Container testsigma_mysql       Waiting                                                                                                                                                              145.2s
#--  ⠿ Container testsigma_server      Created                                                                                                                                                                0.1s
#-- container for service "mysql" is unhealthy
#-- [root@CNT7XATST01 docker-compose]-- 


-- ============================================================================
-- ============================================================================

-- step 4
-- 检查结果 testsigma-server 服务端应用

-- 查看镜像列表
docker images
-- 查看容器列表
docker ps
-- 查看网络端口
netstat -nltp
#
-- 执行结果，如下：
#-- [root@CNT7XATST01 docker-compose]-- docker images
#-- REPOSITORY           TAG       IMAGE ID       CREATED        SIZE
#-- testsigmahq/server   v1.1.0    e6f4f1e9653e   4 months ago   854MB
#-- mysql                5.7       c20987f18b13   7 months ago   448MB
#-- [root@CNT7XATST01 docker-compose]-- 
#-- [root@CNT7XATST01 docker-compose]-- docker ps
#-- CONTAINER ID   IMAGE       COMMAND                  CREATED         STATUS                     PORTS                                                    NAMES
#-- 5774da9b92fe   mysql:5.7   "docker-entrypoint.s…"   6 minutes ago   Up 6 minutes (unhealthy)   33060/tcp, 0.0.0.0:49153->3306/tcp, :::49153->3306/tcp   testsigma_mysql
#-- [root@CNT7XATST01 docker-compose]-- 
#-- [root@CNT7XATST01 docker-compose]-- 
#-- [root@CNT7XATST01 docker-compose]-- netstat -nltp
#-- Active Internet connections (only servers)
#-- Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
#-- tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      878/sshd            
#-- tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN      1079/master         
#-- tcp        0      0 0.0.0.0:49153           0.0.0.0:*               LISTEN      2028/docker-proxy   
#-- tcp6       0      0 :::22                   :::*                    LISTEN      878/sshd            
#-- tcp6       0      0 :::49153                :::*                    LISTEN      2036/docker-proxy   
#-- [root@CNT7XATST01 docker-compose]-- 
  
-- ============================================================================
  

ls -al /home/app/testsigma-server/docker-compose
-- 查询结果，如下：
#-- [root@CNT7XATST01 docker-compose]-- pwd
#-- /home/app/testsigma-server/docker-compose
#-- [root@CNT7XATST01 docker-compose]-- ls -al /home/app/testsigma-server/docker-compose
#-- total 8
#-- drwxr-xr-x 3 root    root   47 Aug  4 13:04 .
#-- drwxr-xr-x 3 root    root   28 Aug  4 12:58 ..
#-- drwxr-xr-x 6 polkitd root 4096 Aug  4 13:04 db_data
#-- -rw-r--r-- 1 root    root  895 Aug  4 12:58 docker-compose.yml
#-- [root@CNT7XATST01 docker-compose]--  
 

-- ============================================================================
-- ============================================================================

**Comment by amitsavyasachi** [comment]
@tiandong19860806 this issue has been fixed with the latest update. You can use the latest version and let us know if you are facing this issue. You can remove the older set up completely and do a clean set up
